src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -265,6 +267,7 @@ private static String buildUserAgent() {
     String languageVersion =
         (systemGetProperty("java.version") != null) ? systemGetProperty("java.version") : "";
     builder.append(languageVersion);
+    builder.append(null == customSuffix ? "" : " IDENTIFIER/" + customSuffix);
```

Comment:
We should probably allow the custom suffix to specify the entire user-agent tuple instead of only the "version" part after the slash. Requiring `IDENTIFIER` as the product name is a bit heavy-handed.

Though people often don't conform to the RFC https://www.rfc-editor.org/rfc/rfc7231#section-5.5.3 technically the part after the slash is only for "versioning information". A sub-application really needs the ability to specify the part before the slash too.

Reply:
Thanks for the review! My original thinking was to reserver some leading characters in guid/version space to identify a product and the rest of the characters for their instance. But I agree that it might be more readable and easier to let the user have full control over the suffix. I have made the change to fix it.